modules/postman-templating/src/template-client.ts

```diff
@@ -189,6 +187,22 @@ export class TemplateClient {
      * removeEmptyLines
      */
     if (options.removeEmptyLines) {
+      // Remove empty tags so that collapsing of line breaks will work even if line contains empty tags
+      // Trim all empty spaces between tags so that they can be recognised as empty. E.g. <b> </b>
+      result = result.replace(/>\s+</g, '><')
+      const $ = cheerio.load(result, { xmlMode: true })
+      const emptyElements = (): cheerio.Cheerio =>
+        $(':empty').not('br, hr, img')
+
+      // Repeatedly remove until there are no more empty elements (maximum of 5 levels).
+      // E.g. After first pass <b><i></i><b> will become <b></b>.Therefore we will need another pass.
+      let levels = 1
+      while (emptyElements().length > 0 && levels <= 5) {
```

Comment:
should this be recursive?

Reply:
Yeah, I think a more efficient/better implementation might be to manually traverse the entire DOM tree and recursively remove nodes that are empty, compared to repeatedly calling the selector (which I assume does a full traversal again each time it's called). Let me try to see if I can implement that.